testgrid/cmd/summarizer/summarizer_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"testing"
+	"time"
+
+	responsepb "k8s.io/test-infra/testgrid/cmd/summarizer/response"
+	configpb "k8s.io/test-infra/testgrid/config"
+	summarypb "k8s.io/test-infra/testgrid/summary"
+)
+
+func TestTableToSummary(t *testing.T) {
+	mockTestRows := []*responsepb.Row{
```

Comment:
I think that this function is just constructing a proto, but is this sufficient to test the function?

What does summarizer do if there was no latest green test at all?

Reply:
In proto3, all fields are optional, and our structs give default values. Currently, the translation is lenient that we are just testing a single DashboardTabSummary output.

So if latest green is missing from the input, it will default to empty string in both the proto and the output object. The same asserts will pass.

Reply:
Expanded the test to cover all output fields.